x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsResultProcessor.java

```diff
@@ -93,5 +108,56 @@ private void processResult(AnalyticsResult result, DataFrameRowsJoiner resultsJo
         if (progressPercent != null) {
             progressTracker.analyzingPercent.set(progressPercent);
         }
+        TrainedModelDefinition inferenceModel = result.getInferenceModel();
+        if (inferenceModel != null) {
+            createAndIndexInferenceModel(inferenceModel);
+        }
+    }
+
+    private void createAndIndexInferenceModel(TrainedModelDefinition inferenceModel) {
+        TrainedModelConfig trainedModelConfig = createTrainedModelConfig(inferenceModel);
+        CountDownLatch latch = storeTrainedModel(trainedModelConfig);
+
+        try {
+            if (latch.await(30, TimeUnit.SECONDS) == false) {
```

Comment:
I think it would be a good idea to write an audit message saying that the trained model was created and its ID.

Reply:
Well spotted. I added this as well as an error audit if storing the model fails.